File: src/BaseHmiTypes/Screens/Controls/HmiSystemDiagnosisColumn.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiSystemDiagnosisColumn
{
    public HmiSystemDiagnosisColumnType Type { get; set; }

    public string? SourceType { get; set; }

    public HmiProperty<bool>? Visible { get; set; }

    public HmiProperty<double>? Width { get; set; }

    public HmiMultilingualText? HeaderText { get; set; }

    public HmiProperty<HmiHorizontalAlignment>? Alignment { get; set; }

    public string? Format { get; set; }

    public HmiProperty<int>? Order { get; set; }
}

File: typescript/base-hmi-types/src/screens/controls/HmiSystemDiagnosisColumn.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiHorizontalAlignment, HmiProperty } from "../base.js";
import { HmiSystemDiagnosisColumnType } from "./HmiSystemDiagnosisColumnType.js";

export class HmiSystemDiagnosisColumn {
  type = HmiSystemDiagnosisColumnType.Unknown;
  sourceType?: string;
  visible?: HmiProperty<boolean>;
  width?: HmiProperty<number>;
  headerText?: HmiMultilingualText;
  alignment?: HmiProperty<HmiHorizontalAlignment>;
  format?: string;
  order?: HmiProperty<number>;
}
